actors/test/committed_capacity_scenario_test.go

```diff
@@ -0,0 +1,352 @@
+	addVerifierParams := verifreg.AddVerifierParams{
+		Address:   verifier,
+		Allowance: abi.NewStoragePower(32 << 40),
+	}
+	_, code = v.ApplyMessage(vm.VerifregRoot, builtin.VerifiedRegistryActorAddr, big.Zero(), builtin.MethodsVerifiedRegistry.AddVerifier, &addVerifierParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	addClientParams := verifreg.AddVerifiedClientParams{
+		Address:   verifiedClient,
+		Allowance: abi.NewStoragePower(32 << 40),
+	}
+	_, code = v.ApplyMessage(verifier, builtin.VerifiedRegistryActorAddr, big.Zero(), builtin.MethodsVerifiedRegistry.AddVerifiedClient, &addClientParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// add market collateral for clients and miner
+	collateral := big.Mul(big.NewInt(3), vm.FIL)
+	_, code = v.ApplyMessage(unverifiedClient, builtin.StorageMarketActorAddr, collateral, builtin.MethodsMarket.AddBalance, &unverifiedClient)
+	require.Equal(t, exitcode.Ok, code)
+	_, code = v.ApplyMessage(verifiedClient, builtin.StorageMarketActorAddr, collateral, builtin.MethodsMarket.AddBalance, &verifiedClient)
+	require.Equal(t, exitcode.Ok, code)
+	collateral = big.Mul(big.NewInt(64), vm.FIL)
+	_, code = v.ApplyMessage(worker, builtin.StorageMarketActorAddr, collateral, builtin.MethodsMarket.AddBalance, &minerAddrs.IDAddress)
+	require.Equal(t, exitcode.Ok, code)
+
+	// create 3 deals, some verified and some not
+	dealIDs := []abi.DealID{}
+	dealStart := v.GetEpoch() + miner.MaxProveCommitDuration[sealProof]
+	deals := publishDeal(t, v, worker, verifiedClient, minerAddrs.IDAddress, "deal1", 1<<30, true, dealStart, 181*builtin.EpochsInDay)
+	dealIDs = append(dealIDs, deals.IDs...)
+	deals = publishDeal(t, v, worker, verifiedClient, minerAddrs.IDAddress, "deal2", 1<<32, true, dealStart, 200*builtin.EpochsInDay)
+	dealIDs = append(dealIDs, deals.IDs...)
+	deals = publishDeal(t, v, worker, unverifiedClient, minerAddrs.IDAddress, "deal3", 1<<34, false, dealStart, 210*builtin.EpochsInDay)
+	dealIDs = append(dealIDs, deals.IDs...)
+
+	//
+	// Precommit, Prove, Verify and PoSt committed capacity sector
+	//
+
+	// precommit capacity upgrade sector with deals
+	ccSectorNumber := abi.SectorNumber(101)
+	ccSealedCid := tutil.MakeCID("101", &miner.SealedCIDPrefix)
+	preCommitParams = miner.SectorPreCommitInfo{
+		SealProof:              sealProof,
+		SectorNumber:           ccSectorNumber,
+		SealedCID:              ccSealedCid,
+		SealRandEpoch:          v.GetEpoch() - 1,
+		DealIDs:                dealIDs,
+		Expiration:             v.GetEpoch() + 220*builtin.EpochsInDay,
+		ReplaceCapacity:        true,
+		ReplaceSectorDeadline:  dlInfo.Index,
+		ReplaceSectorPartition: pIdx,
+		ReplaceSectorNumber:    sectorNumber,
+	}
+
+	// prove commit, verify and PoSt
+	_, code = v.ApplyMessage(addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.PreCommitSector, &preCommitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// assert successful precommit invocation
+	none := []vm.ExpectInvocation{}
```

Comment:
Can you use `nil`? I'm guessing `nil` is interpreted as "don't care", but I think that would be fine here.

Reply:
I started out leaving subinvocations at the leaves untested, but I wasn't sure whether the leaves were actually leaves. This guarantees there are no unexpected calls between actors.